### DISCUSSION TITLE
Add support for Photometric stereo (PS)-AD dataset

### DISCUSSION THREAD

**Discussion by jpcbertoldo** [open-edge-platform/anomalib#649] [Feature Requests]
[1] recently published a dataset similar to MVTec-AD with photometric stereo images -- each object has 4 images with exactly the same position  but with different illumination angles (example below).

![image](https://user-images.githubusercontent.com/24547377/197533921-044080d4-4117-4310-91c4-7836b306b2e6.png)

**Describe the solution you'd like**

Create a dataset and datamodule for this dataset.

This should probably wait for #572.

**Additional context**

~The paper (version from the 18 October 2022, link below) says (footnote 1) "Dataset and codes **will be available** when this paper becomes public.", so I believe it has not yet been published.~

[update] 

The paper's code is available here: [github.com/Junyong-Jung/PSAD](https://github.com/Junyong-Jung/PSAD)

The dataset is availbla for download here (~100Gb): [cnu365-my.sharepoint.com/:u:/g/personal/jjy9385_o_cnu_ac_kr/EeAfcf2zU9NCoz8KjhpeJA8BA5Xv6gp0yxryw-o5AsSUXw](https://cnu365-my.sharepoint.com/:u:/g/personal/jjy9385_o_cnu_ac_kr/EeAfcf2zU9NCoz8KjhpeJA8BA5Xv6gp0yxryw-o5AsSUXw)

Size of the train/test sets for the 10 categories in the dataset (numbers in terms of instances of object, the number of images is multiplied by 4).

![image](https://user-images.githubusercontent.com/24547377/197560114-e88bf933-fc06-4463-aa0d-07647ae962ad.png)

Source: [1].

---

1. Jung, J., Han, S., Park, J. & Cho, D. A Comprehensive Real-World Photometric Stereo Dataset for Unsupervised Anomaly Detection. IEEE Access 10, 108914–108923 (2022).

Link to the paper on Semantic Scholar: [www.semanticscholar.org/paper/Anomaly-Detection-Requires-Better-Representations-Reiss-Cohen/f785d513eb6cef58de1e761534a1598515118faa](https://www.semanticscholar.org/paper/Anomaly-Detection-Requires-Better-Representations-Reiss-Cohen/f785d513eb6cef58de1e761534a1598515118faa)

**Comment by samet-akcay** [comment]
Thanks @jpcbertoldo, looking promising! Moving to feature requests for now.
